src/runtime/server/middleware/session/index.ts

```diff
@@ -81,24 +118,31 @@ const getSession = async (event: H3Event): Promise<null | Session> => {
     return null
   }
 
-  // 3. Is the session not expired?
-  const sessionExpiryInSeconds = useRuntimeConfig().session.session.expiryInSeconds
-  if (sessionExpiryInSeconds !== null) {
-    const now = dayjs()
-    if (now.diff(dayjs(session.createdAt), 'seconds') > sessionExpiryInSeconds) {
-      return null
+  const runtimeConfig = useRuntimeConfig()
+  const sessionOptions = runtimeConfig.session.session as SessionOptions
+
+  try {
+    // 3. Is the session not expired?
+    checkSessionExpirationTime(session, sessionOptions)
+
+    // 4. Check for IP pinning logic
+    await checkSessionIp(event, session, sessionOptions)
```

Comment:
The idea of keeping `if` here was to let other developer scope their mental maps. If they see `if (sessionOptions.ipPinning)` here, they don't have to navigate to the `checkSessionIp` function to understand what side effects it might introduce